beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -806,8 +831,25 @@ where
                                             request,
                                         };
                                     let delay_key = &entry.get().1;
-                                    self.outbound_substreams_delay
-                                        .reset(delay_key, Duration::from_secs(RESPONSE_TIMEOUT));
+                                    let remaining_number_of_chunks_for_this_stream: u64 =
+                                        match &entry.get().3 {
```

Comment:
This could be written a bit more succinctly and we should always be careful of underflows. 
Maybe something like this:
```rust
let remaining_number_of_chunks_for_this_stream = entry.get().3.map(|count| count.saturating_sub(1)).unwrap_or_else(|| 0);
```

Reply:
Nice use of the `unwrap_or_else`. thanks!